huaweicloud/services/secmaster/data_source_huaweicloud_secmaster_alerts.go

```diff
@@ -0,0 +1,1038 @@
+func dataObjectRemediationElem() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"url": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The link to the general fix information for the incident.`,
+			},
+			"recommendation": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The recommended solution.`,
+			},
+		},
+	}
+}
+
+func dataObjectFileInfoElem() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"file_path": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The file path.`,
+			},
+			"file_content": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The file content.`,
+			},
+			"file_new_path": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The file new path.`,
+			},
+			"file_hash": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The file hash value.`,
+			},
+			"file_md5": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The file MD5 value.`,
+			},
+			"file_sha256": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The file SHA256 value.`,
+			},
+			"file_attr": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The file attribute.`,
+			},
+		},
+	}
+}
+
+func dataSourceAlertsRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	var mErr *multierror.Error
+
+	// getAlerts: Query the SecMaster alerts
+	var (
+		listAlertsHttpUrl = "v1/{project_id}/workspaces/{workspace_id}/soc/alerts/search"
+		listAlertsProduct = "secmaster"
+	)
+	client, err := cfg.NewServiceClient(listAlertsProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating SecMaster client: %s", err)
+	}
+
+	listAlertsPath := client.Endpoint + listAlertsHttpUrl
+	listAlertsPath = strings.ReplaceAll(listAlertsPath, "{project_id}", client.ProjectID)
+	listAlertsPath = strings.ReplaceAll(listAlertsPath, "{workspace_id}", fmt.Sprintf("%v", d.Get("workspace_id")))
+
+	listAlertsOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+
+	bodyParams, err := buildAlertBodyParams(d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	alerts := make([]interface{}, 0)
+	offset := 0
+	for {
+		bodyParams["offset"] = offset
+		listAlertsOpt.JSONBody = bodyParams
+		listAlertsResp, err := client.Request("POST", listAlertsPath, &listAlertsOpt)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		listAlertsRespBody, err := utils.FlattenResponse(listAlertsResp)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		data := utils.PathSearch("data", listAlertsRespBody, make([]interface{}, 0)).([]interface{})
+		alerts = append(alerts, data...)
+
+		if len(data) < 1000 {
+			break
+		}
+		offset += 1000
```

Comment:
Is the total in return is valid? when the total is exactly 1000, no need to get next page.